crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -1238,30 +1271,30 @@ impl ScheduleGraph {
             .zip(schedule.systems.drain(..))
             .zip(schedule.system_conditions.drain(..))
         {
-            self.systems[id.index()].inner = Some(system);
-            self.system_conditions[id.index()] = conditions;
+            self.systems[id.0].inner = Some(system);
```

Comment:
I think that reintroducing an `index` method makes sense. See comments above about reader methods.